http-service-mock/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+#![cfg_attr(feature = "nightly", deny(missing_docs))]
+#![feature(futures_api, async_await, await_macro, arbitrary_self_types)]
+
+use futures::{executor::block_on, prelude::*};
+use http_service::{HttpService, Request, Response};
+
+pub struct TestBackend<T: HttpService> {
+    service: T,
+    connection: T::Connection,
+}
+
+impl<T: HttpService> TestBackend<T> {
+    fn wrap(service: T) -> Result<Self, <T::ConnectionFuture as TryFuture>::Error> {
```

Comment:
Or simply you can make this method `pub` and make crate consumers use this instead of `make_server` 😉